streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -200,6 +200,9 @@ public StateStore getStateStore(final String name) {
             }
 
             final String sendTo = toInternal.child();
+            if (currentNode() == null) {
+                throw new StreamsException("Current node is unknown when forwarding to: " + key);
```

Comment:
Why do we add the `key` ? Does not seem to be useful and could potentially leak data into logs.

We should also add a sentence about a potential root cause (even if we add other guards for it already in this PR):
```
throw new StreamsException("Current node is unknown. This can happen if `forward()` is called in an illegal scope. The root cause could be that a `Processor` or `Transformer` instance is shared. To avoid this error, make sure that your suppliers return new instances each time `get()` is called and do not return the same object reference multiple times.");
```

Reply:
Made the change!